Issue# 306: Added tagging controller (#308)
kubernetes/cloud-provider-aws: docs/tagging_controller.md

# The Tagging Controller

> The tagging controller is responsible for tagging and untagging node resources when it joins and leaves the cluster respectively. It can add and remove tags based on user input. Unlike the existing controllers, the tagging controller is working exclusively with AWS as we want to tag the resources (EC instances for example). For functionalities used by the controller, we primarily use `CreateTags` and `DeleteTags` from `EC2`.

Reviewer: ```suggestion
The tagging controller is responsible for tagging and untagging node resources when they join and leave the cluster, respectively. It can add and remove tags based on user input. Unlike the existing controllers, the tagging controller works exclusively with AWS. The AWS APIs it uses are `ec2:CreateTags` and `ec2:DeleteTags`.
```

| Flag | Valid Values | Default | Description |
|------| --- | --- | --- |

> | tags          | Comma-separated list of key=value | -   | A comma-separated list of key-value pairs which will be recorded as nodes' additional tags. For example: "Key1=Val1,Key2=Val2,KeyNoVal1=,KeyNoVal2" |

Reviewer 2: `KeyNoVal1=,KeyNoVal2` does this look right?

Author: I'd assume both `KeyNoVal1=` and `KeyNoVal2` are the right format for keys with no value in cli. @nckturner pls let me know it that's not correct.
